### PR TITLE
Refactor submission

### LINKED ISSUE #22782: Normalize behaviour of space vs. return for activating elements
Right now, `Document::dispatch_key_event` triggers different behaviour for elements if you activate them with spacebar or return. Spacebar performs a [synthetic click activation](https://html.spec.whatwg.org/multipage/interaction.html#activation:fire-a-click-event), while Return triggers [implicit form submission](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#implicit-submission). This means that pressing Return with an anchor element active does not navigate to the link, and special cases the behaviour for text input elements and buttons. I propose the following:
* remove the `implicit_submission` method from the `Activatable` trait
* remove the `implicit_submission` implementation for `HTMLButtonElement` entirely
* make the `implicit_submission` implementation for `HTMLInputElement` a regular method
* treat the Space and Return cases the same in `dispatch_key_event`; remove any reference to implicit submission

This should allow implicit submissions of forms from text inputs to continue working, since there is code in HTMLInputElement::handle_event to recognize the Return key already. Activating buttons with the Return key should also work as expected, since they will receive a synthetic click event.

Code: `components/script/dom/activation.rs`, `components/script/dom/document.rs`, `components/script/dom/htmlinputelement.rs`, `components/script/htmlbuttonelement.rs`

Some tests that might be relevant:
* `./mach test-wpt tests/wpt/web-platform-tests/html/semantics/forms/the-input-element/`
* `./mach test-wpt tests/wpt/web-platform-tests/html/semantics/forms/attributes-common-to-form-controls/`
* `./mach test-wpt tests/wpt/web-platform-tests/html/semantics/forms/form-submission-0/`

### PR DESCRIPTION
**Pull request by garasubo**
This is my first contribution, so please correct me if I'm wrong

I refactored the code as described #22782.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [ ] These changes fix #22782  (GitHub issue number if applicable)

- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because ___

### CODE REVIEW

**Comment by garasubo** [comment]
I didn't add any tests related to this. Do I need to add any tests related to this?

**Comment by garasubo** [comment]
@nox Is there any steps I should do to get it merged?

**Comment by nox** [comment]
Nope, just way for a review. You filed it on Friday and then it was time for the weekend.

File: components/script/dom/document.rs [hunk 1]
```diff
@@ -1469,35 +1469,20 @@ impl Document {
...
-                            el,
-                            false,
-                            false,
-                            false,
-                            false,
-                            ActivationSource::NotFromClick,
-                        )
-                    }
+            if (keyboard_event.key == Key::Enter && keyboard_event.state == KeyState::Up) ||
+                (keyboard_event.code == Code::Space && keyboard_event.state == KeyState::Down)
```
Comment (nox) [inline components/script/dom/document.rs]: In the code before your change, it was `KeyState::Up` for both spaces and enters, is it intended that you now check for `KeyState::Down` for spaces?

**Review by nox — COMMENTED** [review]
Got a single question, otherwise it looks good.

File: components/script/dom/document.rs (hunk 1 above)
Comment (garasubo) [reply components/script/dom/document.rs]: Oh, no. It's not intended. Fixed.

**Comment by nox** [comment]
Please squash the commits together and I'll approve the PR.

**Comment by garasubo** [comment]
Squashed the commits.

**Comment by nox** [comment]
Thanks!

@bors-servo r+

**Comment by CYBAI** [comment]
@bors-servo retry

- network issue in magicleap and android builds

**Pull request merged**